### PR TITLE
ref(flags): instrument sdk featureFlagsIntegration to track FE flag evals

### PR DESCRIPTION
**Pull request by aliu39**
Re-opening https://github.com/getsentry/sentry/pull/81159/files after releasing a custom flag tracking integration in JS SDK 8.43. WIth this integration, we don't need to manage our own buffer in featureObserver anymore. Sentry employees shouldn't notice any changes, this is a proof-of-concept for the new SDK.

The SDK has its own tests already, so I'm only unit testing the add(__)FeaturesHook fxs.

Manually tested with dev-ui sending to a test org: https://andrew-onboarding.sentry.io/share/issue/664703eb3a98441b8a203dc44e803cee/

### CODE REVIEW

File: static/app/utils/featureFlags.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    );
+  if (!featureFlagsIntegration || !('addFeatureFlag' in featureFlagsIntegration)) {
+    logger.error(
+      'Unable to track flag evaluations because FeatureFlagsIntegration is not installed correctly.'
+    );
+    return (_name, _value) => {};
+  }
+  return (name: string, value: unknown) => {
+    // Append `feature.organizations:` in front to match the Sentry options automator format
+    featureFlagsIntegration?.addFeatureFlag('feature.organizations:' + name, value);
```
Comment (billyvg) [inline static/app/utils/featureFlags.ts:25]: Isn't this used for projects and orgs?

Comment (aliu39) [reply static/app/utils/featureFlags.ts:25]: Shoot, yeah

File: static/app/utils/featureFlags.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+import type {Organization} from 'sentry/types/organization';
+import type {Project} from 'sentry/types/project';
+
+/**
+ * Returns a callback that can be used to track sentry flag evaluations through
+ * the Sentry SDK, in the event context. If the FeatureFlagsIntegration is not
+ * installed, the callback is a no-op.
+ */
+export function getSentryFeaturesHook(): (name: string, value: unknown) => void {
```
[the remark this answers is not in the document]
Comment (aliu39) [reply static/app/utils/featureFlags.ts]: wdym? "hook" is the term I'm using to describe the callback taking in a flag name/val. This term is used sometimes in SDK setup code

[the remark this answers is not in the document]
Comment (billyvg) [reply static/app/utils/featureFlags.ts]: sorry this is just semantics but you're not really returning a hook, but rather a handler for your hook. the "hook" is when `features.includes()` gets called.

[the remark this answers is not in the document]
Comment (aliu39) [reply static/app/utils/featureFlags.ts]: ok got it, what if I rename all instances of "hook" to "handler"?

**Review by billyvg — APPROVED** [review]

**Pull request merged**